### PR TITLE
FOGL-2820 Add installation of Python depenencies in post install script

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: plugins/packages/RPM/SPECS/plugin.spec [hunk 1]
```diff
@@ -121,4 +121,7 @@ set_files_ownership () {
 echo "Setting ownership of FogLAMP files"
 set_files_ownership
 
+# Install any Python dependencies
+find /usr/local/foglamp/__INSTALL_DIR__ -name "requirements*.txt" -exec pip3 install -Ir "{}" --no-cache-dir \;
```
Comment (ashish-jabble) [inline plugins/packages/RPM/SPECS/plugin.spec]: As we packaged requirements file under `/usr/local/foglamp/python/` directory. So above line will never execute as we find the file in `/usr/local/foglamp/__INSTALL_DIR__`.
**For example**:
```
/usr/local/foglamp/python/foglamp/plugins/south/coap/README.rst
/usr/local/foglamp/python/foglamp/plugins/south/coap/__init__.py
/usr/local/foglamp/python/foglamp/plugins/south/coap/coap.py
/usr/local/foglamp/python/requirements-coap.txt
```
So first we need the name of plugin and then find requirements under `/usr/local/foglamp/python`. We may fix with below code OR any better solution

```
plugin_name=`echo __NAME__ | sed 's/.*-//'`

# Install any Python dependencies
find /usr/local/foglamp/python -name "requirements-${plugin_name}.txt" -exec pip3 install -Ir "{}" --no-cache-dir \;
```

**Review by ashish-jabble — CHANGES_REQUESTED** [review]
Discovery of a plugin is not working when install package via `yum`
```
May 20 10:29:45 ip-10-0-0-170 yum[11057]: Installed: foglamp-south-coap-1.5.2-1.x86_64
May 20 10:30:08 ip-10-0-0-170 journal: FogLAMP[818] ERROR: plugin_discovery: foglamp.common.plugin_discovery: Plugin "coap" import problem from path "foglamp.plugins.south". No module named 'aiocoap'
```
Though I made changes in script at local to `find` the requirements file in correct path and which is fine. But now the another problem has been occurred on install package with changes I have made.
```
find: ‘pip3’: No such file or directory
  Verifying  : foglamp-south-coap-1.5.2-1.x86_64  
```

File: plugins/packages/RPM/SPECS/plugin.spec [hunk 2]
```diff
@@ -121,4 +121,9 @@ set_files_ownership () {
 echo "Setting ownership of FogLAMP files"
 set_files_ownership
 
+# Install any Python dependencies
+if [ -f /usr/local/foglamp/python/requirements-__PLUGIN_NAME__.txt ]; then
+	pip3 install -Ir /usr/local/foglamp/python/requirements-__PLUGIN_NAME__.txt
```
Comment (ashish-jabble) [inline plugins/packages/RPM/SPECS/plugin.spec]: It is not working as pip3 command is not found, attached below O/P
```
Running transaction
  Installing : foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64                                                                                 1/1 
Setting ownership of FogLAMP files
/var/tmp/rpm-tmp.MEO49J: line 42: pip3: command not found
warning: %post(foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64) scriptlet failed, exit status 127
Non-fatal POSTIN scriptlet failure in rpm package foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64
  Verifying  : foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64  
```

**Potential fix**
```
 if [ -f /usr/local/foglamp/python/requirements-__PLUGIN_NAME__.txt ]; then
-       pip3 install -Ir /usr/local/foglamp/python/requirements-__PLUGIN_NAME__.txt
+       su - <<EOF
+scl enable rh-python36 bash
+pip install -Ir /usr/local/foglamp/python/requirements-__PLUGIN_NAME__.txt
+EOF
fi
```
ONLY With above change I am able to install package successfully. Below is the console log output and plugin of discovery is okay and service has been added successfully

**Console logs**

```
sudo yum install archive/x86_64/foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64.rpm 
Loaded plugins: amazon-id, rhui-lb, search-disabled-repos
Examining archive/x86_64/foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64.rpm: foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64
Marking archive/x86_64/foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package foglamp-south-mqtt_sparkplug.x86_64 0:1.5.2-1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

===============================================================================================================================================
 Package                                 Arch              Version               Repository                                               Size
===============================================================================================================================================
Installing:
 foglamp-south-mqtt_sparkplug            x86_64            1.5.2-1               /foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64             73 k

Transaction Summary
===============================================================================================================================================
Install  1 Package

Total size: 73 k
Installed size: 73 k
Is this ok [y/d/N]: y 
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64                                                                                 1/1 
Setting ownership of FogLAMP files
Collecting paho-mqtt==1.4.0 (from -r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/25/63/db25e62979c2a716a74950c9ed658dce431b5cb01fde29eb6cba9489a904/paho-mqtt-1.4.0.tar.gz (88kB)
Collecting google==2.0.2 (from -r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/34/4c/9bc51ae2611e5893ff45f8972f20dd7c8408eb5d706a541182ac2da3b0b7/google-2.0.2.tar.gz (45kB)
Collecting protobuf==3.7.0 (from -r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/c5/60/ca38e967360212ddbb004141a70f5f6d47296e1fba37964d8ac6cb631921/protobuf-3.7.0-cp36-cp36m-manylinux1_x86_64.whl (1.2MB)
Collecting beautifulsoup4 (from google==2.0.2->-r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/1d/5d/3260694a59df0ec52f8b4883f5d23b130bc237602a1411fa670eae12351e/beautifulsoup4-4.7.1-py3-none-any.whl (94kB)
Collecting setuptools (from protobuf==3.7.0->-r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/ec/51/f45cea425fd5cb0b0380f5b0f048ebc1da5b417e48d304838c02d6288a1e/setuptools-41.0.1-py2.py3-none-any.whl (575kB)
Collecting six>=1.9 (from protobuf==3.7.0->-r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting soupsieve>=1.2 (from beautifulsoup4->google==2.0.2->-r /usr/local/foglamp/python/requirements-mqtt_sparkplug.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/b9/a5/7ea40d0f8676bde6e464a6435a48bc5db09b1a8f4f06d41dd997b8f3c616/soupsieve-1.9.1-py2.py3-none-any.whl
Installing collected packages: paho-mqtt, soupsieve, beautifulsoup4, google, setuptools, six, protobuf
  Running setup.py install for paho-mqtt: started
    Running setup.py install for paho-mqtt: finished with status 'done'
  Running setup.py install for google: started
    Running setup.py install for google: finished with status 'done'
Successfully installed beautifulsoup4-4.7.1 google-2.0.2 paho-mqtt-1.4.0 protobuf-3.7.0 setuptools-41.0.1 six-1.12.0 soupsieve-1.9.1
You are using pip version 9.0.1, however version 19.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Last login: Mon May 20 09:35:11 UTC 2019 on pts/0
  Verifying  : foglamp-south-mqtt_sparkplug-1.5.2-1.x86_64                                                                                 1/1 

Installed:
  foglamp-south-mqtt_sparkplug.x86_64 0:1.5.2-1                                                                                                

Complete!

```

Comment (MarkRiddoch) [reply plugins/packages/RPM/SPECS/plugin.spec]: The su in this solution should not be required as it already runs as root.

**Review by ashish-jabble — APPROVED** [review]

**Pull request merged**